app/views/contexts/section_summary_context.py

```diff
@@ -0,0 +1,160 @@
+from typing import Iterator, Mapping
+
+from flask import url_for
+
+from app.questionnaire import QuestionnaireSchema
+from .context import Context
+from .list_context import ListContext
+from .summary import Group
+
+
+class SectionSummaryContext(Context):
+    def __init__(
+        self, language, schema, answer_store, list_store, progress_store, metadata
+    ):
+        super().__init__(
+            language, schema, answer_store, list_store, progress_store, metadata
+        )
+        self.list_context = ListContext(
+            self._language,
+            self._schema,
+            self._answer_store,
+            self._list_store,
+            self._progress_store,
+            self._metadata,
+        )
+
+    def __call__(self, current_location):
+        try:
+            block = self._schema.get_block(current_location.block_id)
+            collapsible = block.get("collapsible", False)
+        except AttributeError:
```

Comment:
What can raise an `AttributeError` here?

Reply:
A block that resolves to None

Reply:
I think this highlights another problem we will get to when we move section summaries out of blocks - where will `collapsible` be configured? I think it would make sense for it to be a property of the `summary` element on a `section`. If we do that we would need to move the list of custom summary elements into another property e.g `items`, something like:

```json
{
  "summary": {
    "collapsible": true,
    "items" : []
  }
}
```

As this is something that we are changing in this pull request, we should consider doing this now.

Reply:
Yep agreed :+1:

Reply:
One question - if a "summary" key is present in a section object, and that key does not contain an array of summary elements, should that then build a default summary view?

Reply:
Yes, I think so.